pkg/aws/ec2/ec2.go

```diff
@@ -43,14 +44,35 @@ type MetricsAPI interface {
 }
 
 // NewClient returns a new EC2 client
-func NewClient(ec2Client *ec2.Client, metrics MetricsAPI, rateLimit float64, burst int) *Client {
+func NewClient(ec2Client *ec2.Client, metrics MetricsAPI, rateLimit float64, burst int, subnetsFilters []ec2.Filter) *Client {
 	return &Client{
-		ec2Client:  ec2Client,
-		metricsAPI: metrics,
-		limiter:    helpers.NewApiLimiter(metrics, rateLimit, burst),
+		ec2Client:      ec2Client,
+		metricsAPI:     metrics,
+		limiter:        helpers.NewApiLimiter(metrics, rateLimit, burst),
+		subnetsFilters: subnetsFilters,
 	}
 }
 
+func NewSubnetsFilters(tags map[string]string, ids []string) []ec2.Filter {
```

Comment:
Please add a comment and a unit test

Reply:
Added both (though github UI don't show the comment as "outdated")